lib/handlers/lnurl_handler.dart

```diff
@@ -24,60 +29,83 @@ class LNURLHandler {
-  var host = Uri.parse(response.callback).host;
-  promptAreYouSure(context, "Open Channel",
-          Text("Are you sure you want to open a channel with $host?"))
-      .then((value) async {
-    if (value) {
-      var synced = await showDialog(
-          context: context,
-          useRootNavigator: false,
-          barrierDismissible: false,
-          builder: (BuildContext context) {
-            return AlertDialog(
-              content: SyncProgressDialog(closeOnSync: true),
-              actions: <Widget>[
-                FlatButton(
-                  child: Text("CANCEL",
-                      style: Theme.of(context).primaryTextTheme.button),
-                  onPressed: () {
-                    Navigator.of(context).pop(false);
-                  },
-                ),
-              ],
-            );
-          });
-      if (synced == true) {
-        var loaderRoute = createLoaderRoute(context);
-        Navigator.of(context).push(loaderRoute);
-        var action = OpenChannel(response.uri, response.callback, response.k1);
-        lnurlBloc.actionsSink.add(action);
-        action.future.catchError((err) {
-          promptError(context, "Open Channel Error",
-              Text("Failed to open channel.\n" + err.toString()));
-        }).whenComplete(() => Navigator.of(context).removeRoute(loaderRoute));
+  void _openLNURLChannel(BuildContext context, LNUrlBloc lnurlBloc,
+      ChannelFetchResponse response) {
+    var host = Uri.parse(response.callback).host;
+    promptAreYouSure(context, "Open Channel",
+            Text("Are you sure you want to open a channel with $host?"))
+        .then((value) async {
+      if (value) {
+        var synced = await showDialog(
+            context: context,
+            useRootNavigator: false,
+            barrierDismissible: false,
+            builder: (BuildContext context) {
+              return AlertDialog(
+                content: SyncProgressDialog(closeOnSync: true),
+                actions: <Widget>[
+                  FlatButton(
+                    child: Text("CANCEL",
+                        style: Theme.of(context).primaryTextTheme.button),
+                    onPressed: () {
+                      Navigator.of(context).pop(false);
+                    },
+                  ),
+                ],
+              );
+            });
+        if (synced == true) {
+          var loaderRoute = createLoaderRoute(context);
+          Navigator.of(context).push(loaderRoute);
+          var action =
+              OpenChannel(response.uri, response.callback, response.k1);
+          lnurlBloc.actionsSink.add(action);
+          action.future.catchError((err) {
+            promptError(context, "Open Channel Error",
+                Text("Failed to open channel.\n" + err.toString()));
+          }).whenComplete(() => Navigator.of(context).removeRoute(loaderRoute));
+        }
       }
+    });
+  }
+
+  void _listenLnLinks() {
```

Comment:
I think it is better to add the loader action to the main stream rather here.
It will be easier to understand.

Reply:
[Done](https://github.com/breez/breezmobile/pull/341/commits/e9fe825edf983b805ed1503de579786af09b6ddc)